### DISCUSSION TITLE
Display speed limit in navigation

### DISCUSSION THREAD

**Discussion by wavard** [devemux86/cruiser#135] [Q&A] [answered]
Which maps will display speedlimits in Cruiser

**Comment by devemux86** [comment]
Do you mean showing speed limits on the route line on the map when calculating a route?

**Comment by wavard** [comment]
Thanks for your reply.  I mean displaying spee limits on the screen while
navigating.

Regards

Walter

**Comment by devemux86** [comment, marked as answer]
Cruiser displays speed limits in navigation, if it finds them in the route.

The speed limits in navigation are provided by the calculated route.
And many services choose to not provide such info in their routes!

Route services that provide speed limits in their routes are currently:
- GraphHopper (offline and online)
- Kurviger (**import** kurviger routes)

**Comment by WolfgangZ42** [comment]
I use [Tempo Master](https://play.google.com/store/search?q=tempo+master&c=apps) as an overlay app since more than 2 years. Mainly because it works in a free ride too. Here an picture:
<img width="600" src="https://user-images.githubusercontent.com/121806238/233836605-c74ad7a5-37b8-4263-862b-8f34029c9378.png"/>

**Comment by wavard** [comment]
Hi Wolfgang
I cannot find Tempo Master.

Regards

Walter

**Comment by wavard** [comment]
Thanks for your reply

Regards

Walter

**Comment by devemux86** [comment]
Probably mean "TempoMaster: GPS Speedometer":

- https://play.google.com/store/apps/details?id=gd.tempomaster_pro

**Comment by umpfmpf** [comment]
Is there a way coming to show the speedlimit in navigation while calculating with brouter?

**Reply by devemux86** [reply]
@umpfmpf 

Unfortunately BRouter does not provide speed limits (or street names).
(related [issue](https://github.com/abrensch/brouter/issues/55))

With BRouter we can use an external app, see above for suggestions.

---

I have ideas to provide speed limits (or street names) inside Cruiser independent of the router,
but more research is needed. :slightly_smiling_face:

**Reply by Ciutnik** [reply]
> I have ideas to provide speed limits (or street names) inside Cruiser independent of the router

Wow 😳 I thought nothing could surprise me anymore... But it's a very very nice surprise! I keep my fingers crossed for success (especially dreaming about street names) 👍

**Reply by umpfmpf** [reply]
i wait for the speed limits. with a motorbike this is very important.

**Reply by WolfgangZ42** [reply]
I'm using TempoMaster since years - it works even off route. But: OSM does not contain all speed limits and sometimes shows even old or wrong ones. Interesting is that TempoMaster uses a downloaded database (for most european countries) and therefore run totally offline. 
![2023-08-17 17_12_01-wenden mp4 - VLC media player](https://github.com/devemux86/cruiser/assets/121806238/0b43fc58-3b1e-4e94-8954-86377adb5cda)
Sorry @Ciutnik ... Poland isn't in the list.

**Reply by Ciutnik** [reply]
> Poland isn't in the list.

It's a pity... But thanks for checking it out. 
Fortunately, we have excellent local services of this type, additionally warning about patrols, traffic jams, roadworks or other events on the road, based on a large community. 

And by the way, I wonder where TempoMaster gets the data, since they are different from OSM. Maybe they are published officially by individual government agencies? After all, this is public data and should be available for everyone.

**Reply by Ciutnik** [reply]
@WolfgangZ42 
Out of curiosity, I looked at the [Tempomaster website](https://www.g-daehling.de/tempomaster_de/index.html). It looks like the data comes from OpenStreetMap 🙂

**Comment by DolomitesTom** [comment]
I only want to see speed limits above 40 km/h exceedance. Speaking of which, since when was the slider limited to 30 km/h? Luckily I still have 40 in there. Can the tolerance be extended to 40 again?

**Reply by devemux86** [reply]
It is just a slider, I can increase it in next update.

@DolomitesTom I wonder how fast are you driving? :slightly_smiling_face:

**Reply by DolomitesTom** [reply]
Only outside built-up areas 🙂 I like to drive fast, only not too fast so that the budget explodes...

... to know at what point it becomes really expensive. Otherwise, if it keeps coming with warnings, I get used to it and end up ignoring it 🙄.

**Reply by devemux86** [reply]
> Can the tolerance be extended to 40 again?

It will be available in next Cruiser.
